File: 724_Find Pivot Index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        # O(n)
        sumNums = sum(nums)
        compNum = 0
        
        # O(n)
        for i in range(len(nums)):
            if compNum == (sumNums - nums[i]) / 2:
                return i
            compNum += nums[i]
            
        return -1

File: Algorithms/724. Find Pivot Index.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var pivotIndex = function (nums) {
  let sum = nums.reduce((acc, curr) => acc + curr, 0);
  let leftSum = 0;

  // T: O(n)
  for (let i = 0; i < nums.length; i++) {
    sum -= nums[i];

    if (leftSum === sum) {
      return i;
    }

    leftSum += nums[i];
  }

  return -1;
};

module.exports = pivotIndex;
